sdk/core/switcher/entry.S

```diff
@@ -347,11 +398,20 @@ exception_entry_asm:
 	csc                ct0, TrustedStack_offset_mepcc(csp)
 	csrr               t1, mstatus
 	csw                t1, TrustedStack_offset_mstatus(csp)
-#ifdef CONFIG_MSHWM
+#ifdef CHERIOT_HAS_MSHWM
 	csrr               t1, CSR_MSHWM
 	csw                t1, TrustedStack_offset_mshwm(csp)
 	csrr               t1, CSR_MSHWMB
 	csw                t1, TrustedStack_offset_mshwmb(csp)
+#endif
+#ifdef CHERIOT_HAS_ZTOP
```

Comment:
You may as well put this `ifdef` underneath the one for `CHERIOT_HAS_MSHWM`, too, since the feature is dependent.